chore: add serverComponentsExternalPackages callout for Next.js (#7757)
aws-amplify/docs: src/pages/gen1/[platform]/build-a-backend/server-side-rendering/nextjs/index.mdx

</Callout>

<Callout>

> **NOTE:** If you used the Amplify server-side APIs in server action and encounter the error running `next build`:

Reviewer: Need to apply the same changes here, sorry

```suggestion
**NOTE:** If you use the Amplify server-side APIs in a server action and encounter the following error running `next build`:
```

> **NOTE:** If you use the Amplify server-side APIs in a server action and encounter the following error running `next build`:

Reviewer 2: ```suggestion
**Note:** If you use the Amplify server-side APIs in a server action and encounter the following error running `next build`:
```

> ./node_modules/@aws-amplify/core/node_modules/@aws-crypto/sha256-js/build/module/index.js + 12 modules

> Cannot get final name for export 'fromUtf8' of ./node_modules/@smithy/util-utf8/dist-es/index.js

You can add the following to your `next.config.js`:

```ts title="next.config.js"
/** @type {import('next').NextConfig} */
const nextConfig = {
  // highlight-start
  experimental: {
    serverComponentsExternalPackages: ['@aws-crypto'],
  },
  // highlight-end
};
```

> See Next.js documentation on [`serverComponentsExternalPackages`](https://nextjs.org/docs/app/api-reference/next-config-js/serverComponentsExternalPackages) fro more details.

Author: ```suggestion
See Next.js documentation on [`serverComponentsExternalPackages`](https://nextjs.org/docs/app/api-reference/next-config-js/serverComponentsExternalPackages) for more details.
```
